src/Refinement/RefinedTriangulations.jl

```diff
@@ -0,0 +1,126 @@
+  get_glue(t.trian,Val(d))
+end
+
+function Base.view(t::RefinedTriangulation,ids::AbstractArray)
+  v = view(t.trian,ids)
+  return RefinedTriangulation(v,t.model)
+end
+
+# Wrap constructors for RefinedDiscreteModel
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,filter::AbstractArray) where d
+  
+  trian = Triangulation(ReferenceFE{d},get_model(model),filter)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,labels::FaceLabeling;kwargs...) where d
+  trian = Triangulation(ReferenceFE{d},get_model(model),labels;kwargs...)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(trian::RefinedTriangulation,args...;kwargs...)
+  return RefinedTriangulation(Triangulation(trian.trian),trian.model)
+end
+
+# Domain changes
+# TODO: This assumes we have the same type of triangulation on both refinement levels!
+#       we might want to change this in the future when doing hybrid methods etc...
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  (strian === ttrian) && (return true)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
+  smodel = get_refined_model(strian)
+  tmodel = get_background_model(ttrian)
+  a = get_model(smodel) === tmodel  # It is fundamentally the same model
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::T,ttrian::RefinedTriangulation) where {T <: Triangulation}
+  return is_change_possible(ttrian,strian)
+end
+
+function Geometry.best_target(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  @check is_change_possible(strian,ttrian)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_model(smodel) === get_parent(tmodel)
+  b = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  a || b ? ttrian : strian
+end
+
+function Geometry.best_target(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
+  @check is_change_possible(strian,ttrian)
+  return strian
+end
+
+function Geometry.best_target(strian::T,ttrian::RefinedTriangulation) where {T <: Triangulation}
+  @check is_change_possible(strian,ttrian)
+  return ttrian
+end
+
+
+"""
+  Given a RefinedTriangulation and a CellField defined on the parent(coarse) mesh, 
+  returns an equivalent CellField on the fine mesh.
+"""
+function change_domain_c2f(f_coarse, ftrian::RefinedTriangulation{Dc,Dp}) where {Dc,Dp}
+  model  = get_refined_model(ftrian)
+  glue   = get_glue(model)
+  if (num_cells(ftrian) != 0)
```

Comment:
In this piece of code we are assuming that `num_cells(ftrian)==length(glue.f2c_faces_map[Dc+1])`. 

I think we should generalize the code so that we do not have this constraint.

Reply:
The only way the condition is false is by having `ftrian` be a view, right? Or am I missing something?

Reply:
> The only way the condition is false is by having ftrian be a view, right? Or am I missing something?

It may also happen if one processor does not have cells at all. This, e.g., happens when we consider a new subset of processors before redistributing the current mesh in the hierarchy (all these new processors do not have cells) . I had to put that condition in the original code as the code within the `if` crashes is the triangulation is void. 

Anyway, my comment above (https://github.com/gridap/Gridap.jl/pull/838#discussion_r1001618449) was not actually related to this `if` statement. Did you understand what I meant?